server/src/main/java/org/elasticsearch/index/mapper/GeoPointFieldMapper.java

```diff
@@ -260,9 +262,13 @@ public Query geoShapeQuery(Geometry shape, String fieldName, ShapeRelation relat
             if (luceneGeometries.length == 0) {
                 return new MatchNoDocsQuery();
             }
-            Query query = LatLonPoint.newGeometryQuery(fieldName, relation.getLuceneRelation(), luceneGeometries);
+            // For point queries and intersects, lucene does not match points that are encoded to Integer.MAX_VALUE. Use contains
+            // instead that returns the expected results.
+            ShapeField.QueryRelation luceneRelation = relation == ShapeRelation.INTERSECTS && shape.type() == ShapeType.POINT ?
+                    ShapeField.QueryRelation.CONTAINS : relation.getLuceneRelation();
```

Comment:
I think in this case explicit if statement might make it much easier to comprehend. There is just too much going on with the comment and expression otherwise. WDYT?
```
final ShapeField.QueryRelation luceneRelation;
if (shape.type() == ShapeType.POINT && relation == ShapeRelation.INTERSECTS) {
  // For point queries and intersects, lucene does not match points that are encoded to Integer.MAX_VALUE. Use contains instead
  luceneRelation = ShapeField.QueryRelation.CONTAINS;
} else {
  luceneRelation = relation.getLuceneRelation();
}

Reply:
done